streams/src/main/java/org/apache/kafka/streams/kstream/internals/OffsetResetStrategy.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+public enum OffsetResetStrategy {
```

Comment:
Maybe we could also re-use `AutoOffsetResetStrategy.StrategyType`?

The weird thing is, that I realized that the exiting `Topology.AutoOffsetReset` does only have EARLIEST and LATEST, but not NONE... NONE is most likely not super useful and nobody asked for it yet, so maybe that's ok. It's just a question of re-using ``AutoOffsetResetStrategy.StrategyType` is the right thing, if we don't allow for NONE -- or maybe it does not matter.

It's also all internal code, we we can change it at any point anyway.

Reply:
\cc @omkreddy for another small KIP update

Reply:
Will add this to https://github.com/apache/kafka/pull/17973

Reply:
just to clarify, the conclusion is that we'll add a `NONE` option to `Topology.AutoOffsetReset` as part of the KIP, and then use `AutoOffsetResetStrategy.StrategyType` internally?

Reply:
The KIP deprecates `Topology.AutoOffsetReset` in favor of `AutoOffsetReset` class -- we will add a static builder `none()` to this new class for public usage.

And yes, we will use `AutoOffsetResetStrategy.StrategyType` internally (instead of adding our own `OffsetResetStrategy` enum) -- added this to the other PR already.

Reply:
ah, thanks for clarifying. yeah that seems like the right call